src/backend/executor/spi.c

```diff
@@ -1294,6 +1294,19 @@ SPI_cursor_open_internal(const char *name, SPIPlanPtr plan,
 	_SPI_current->processed = 0;
 	_SPI_current->tuptable = NULL;
 
+	bool orig_gp_enable_gpperfmon = gp_enable_gpperfmon;
+
+PG_TRY();
+{
+	/*
+	* Temporarily disable gpperfmon since we don't send information for internal queries in
+	* most cases, except when the debugging level is set to DEBUG4 or DEBUG5.
+	*/
+	if (log_min_messages > DEBUG4)
```

Comment:
At the ticket said, that logs should be printed only if log level is DEBUG5. At your case logs will be printed for DEBUG5 and DEBUG4.

Reply:
And yet this is how it is implemented in cc2446b9333c52932c7a4ceeb522966d9cce31c3. I couldn't find any documentation specifying whether the threshold really be DEBUG5 or DEBUG4, so I decided to extend current behavior to cursors.

Reply:
In this case you should add information about the DEBUG4 log level to PR's description.